token/program-2022/Cargo.toml

```
@@ -19,7 +19,7 @@ num-derive = "0.3"
 num-traits = "0.2"
 num_enum = "0.5.4"
 solana-program = "1.9.5"
-solana-zk-token-sdk = "0.6.0"
+solana-zk-token-sdk = "0.7.0"
```

I just published 0.8.0 to pick up this change, https://github.com/solana-labs/solana/pull/23307 that's needed to unblock my PR but 0.7.0/0.8.0 has other interface changes that are in this PR too:
```suggestion
solana-zk-token-sdk = "0.8.0"
```

---

Done.